io/zenoh-transport/src/unicast/establishment/ext/patch.rs

```diff
@@ -0,0 +1,196 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use core::marker::PhantomData;
+
+use async_trait::async_trait;
+use zenoh_buffers::{
+    reader::{DidntRead, Reader},
+    writer::{DidntWrite, Writer},
+};
+use zenoh_codec::{RCodec, WCodec, Zenoh080};
+use zenoh_protocol::transport::init;
+use zenoh_result::Error as ZError;
+
+use crate::unicast::establishment::{AcceptFsm, OpenFsm};
+
+// Extension Fsm
+pub(crate) struct PatchFsm<'a> {
+    _a: PhantomData<&'a ()>,
+}
+
+impl<'a> PatchFsm<'a> {
+    pub(crate) const fn new() -> Self {
+        Self { _a: PhantomData }
+    }
+}
+
+/*************************************/
+/*              OPEN                 */
+/*************************************/
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(crate) struct StateOpen {
+    patch: init::ext::PatchType,
+}
+
+impl StateOpen {
+    pub(crate) const fn new() -> Self {
+        Self {
+            patch: init::ext::PatchType::NONE,
+        }
+    }
+
+    pub(crate) const fn get(&self) -> init::ext::PatchType {
+        self.patch
+    }
+}
+
+#[async_trait]
+impl<'a> OpenFsm for &'a PatchFsm<'a> {
+    type Error = ZError;
+
+    type SendInitSynIn = &'a StateOpen;
+    type SendInitSynOut = init::ext::PatchType;
+    async fn send_init_syn(
+        self,
+        _state: Self::SendInitSynIn,
+    ) -> Result<Self::SendInitSynOut, Self::Error> {
+        Ok(init::ext::PatchType::CURRENT)
+    }
+
+    type RecvInitAckIn = (&'a mut StateOpen, init::ext::PatchType);
+    type RecvInitAckOut = ();
+    async fn recv_init_ack(
+        self,
+        input: Self::RecvInitAckIn,
+    ) -> Result<Self::RecvInitAckOut, Self::Error> {
+        let (state, other_ext) = input;
+        state.patch = other_ext;
+        Ok(())
+    }
+
+    type SendOpenSynIn = &'a StateOpen;
+    type SendOpenSynOut = ();
+    async fn send_open_syn(
+        self,
+        _state: Self::SendOpenSynIn,
+    ) -> Result<Self::SendOpenSynOut, Self::Error> {
+        unimplemented!()
+    }
+
+    type RecvOpenAckIn = (&'a mut StateOpen, ());
+    type RecvOpenAckOut = ();
+    async fn recv_open_ack(
+        self,
+        _state: Self::RecvOpenAckIn,
+    ) -> Result<Self::RecvOpenAckOut, Self::Error> {
+        unimplemented!()
```

Comment:
Add a comment that this is not `unimplemented` but rather no extension exists for that. 
Or should we simply return `Ok`?